[EGD-8148] Download asset documentation update (#3462)
mudita/MuditaOS: doc/CI.md

# Continuous Integration

We are using Jenkins as a CI system.
Each internal PR is checked on the CI, and proper status is added to the PR.

## Checks that are done
Currently we run several checks for each target:
| Check | Description |
| ---- | ----------- |
| Commit Message check | Check if commit message has proper format (see [Development workflow](development_workflow.md)) |
| Copyright notice check | checks if changed files have proper copyright notice |
| Style checking | Runs clang-format with style described in `.clang-format` against committed code |
| Build RT1051 | Compiles code for RT1051 for bell and pure targets |
| Build Linux | Compiles code for Linux, runs Unit Tests, and looks for static variables for bell and pure targets|

## what we have

* all steps are written down in `Jenkinsfile`, these steps use our build library on Jenkins

> * we have:

Reviewer: ```suggestion
* we already have:
```

    * daily builds on CI from which daily releases artifacts are created
    * release builds on CI from which release artifacts are created

>     * ccached builds

Reviewer: ```suggestion
    * ccache'd builds
```

> * drop draft & multiple requests for builds are dropped ( to not choke the builds)

Reviewer: ```suggestion
* drafts & multiple requests for builds are dropped ( to not choke the build's queue)
```
